File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        h = len(nums)
        m = int((h-l)/2)

        while h >= l:
            m = int((h-l)/2)
            if nums[m] == target:
                return m
            elif nums[m] > target:
                l = m - 1
            else:
                h = m + 1
        return -1
                       

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> res;
        map<int, int> mm;
        for (const auto n : nums) {
            if (mm.find(n) != mm.end()) {
                mm[n]++;
            }
            else mm[n] = 1;
        }
        for (int i = 0; i < k; i++) {
            int greatest = 0;
            for (const auto n : nums) {
                if (mm[n] > mm[greatest]) {
                    greatest = n;
                }
            }
            res.push_back(greatest);
            mm[greatest] = -1;
        }
        return res;
    }
};
